server/routes/archiveGoal/archiveGoalController.ts

```diff
@@ -18,18 +19,19 @@ export default class ArchiveGoalController {
 
   getArchiveGoalView: RequestHandler = async (req, res, next): Promise<void> => {
     const { prisonNumber, goalReference } = req.params
-    const { prisonerSummary } = res.locals
+    const { prisonerSummary, allGoalsForPrisoner } = res.locals
 
     let { archiveGoalForm } = getPrisonerContext(req.session, prisonNumber)
     if (!archiveGoalForm || archiveGoalForm.reference !== goalReference) {
-      const actionPlan = await this.educationAndWorkPlanService.getActionPlan(prisonNumber, req.user.username)
-      if (actionPlan.problemRetrievingData) {
+      if (allGoalsForPrisoner.problemRetrievingData) {
```

Comment:
Basically all of the changes in this PR are the same as this ....
Instead of the controller method calling the service method I have changed it so that the route definition (`index.ts` 👇 ) calls the service via a middleware function. The controller is simpler as it can safely assume the middleware has been called and prisoner goals are on `res.locals.allPrisonerGoals`

But in respect of being able to better support when the API starts returning a 404, the change will only be in 1 or 2 places (probably the service method and the middleware)